lib/matplotlib/axes/_base.py

```diff
@@ -3093,32 +3093,22 @@ def draw_artist(self, a):
         """
         Efficiently redraw a single artist.
 
-        This method can only be used after an initial draw of the figure,
-        because that creates and caches the renderer needed here.
         """
-        if self.figure._cachedRenderer is None:
```

Comment:
On one hand, this call making a new renderer would be odd,  my understanding of why this exists is a helper specifcally for managing blitting where you can rely on the render not only existing, but also having already had everything else of interest rasterized.  By caching the renderer on the figure we are saying "yes, this figure was rendered with this renderer so we are sure that it is holding the state of everything else, also you can use it to draw the given artist".  If we cache / create the renderer on the canvas we only get "this renderer will work".

However, that is a narrow case and while I can think of stories of how this would break someone ("catch the `AttirbuteError` to know we need to re-render the whole figure"....but for that to make sense we would also have to be sure that we were properly invalidating the cache (by setting it to None) when we resize the Figure).

Which is a very long way of saying that I am 👍🏻 on this change (despite some risk), will need an API change note.

Reply:
Yes, I agree with all of that. I was a bit surprised that there wasn't more code churn in this PR which does make me a bit hesitant about what the edge-cases I'm missing are...

One thing I thought about is that we have a `get_renderer()` method on Agg/Cairo Canvas' now, but not on PDF, but the PDF requires a file handle so it isn't as straightforward to just point to a renderer. I was curious if that was perhaps the idea behind the figure caching in the first place... But, surprisingly it didn't affect any of the PDF tests.

Reply:
The pdf and svg renderer's work in a streaming mode so going back to one only makes limited sense in general and makes no sense in this case.